### PR TITLE
Support yarn not installed in `create-turbo`

### LINKED ISSUE #241: Setting up via create-turbo should have yarn checks
### Describe the feature you'd like to request

Scenario:
Uninstall yarn completely from the system, and run

```
npx create-turbo@latest yourapp
```

It will fail at the second step where it should install the packages via yarn. Because yarn isn't there.

And it gives an unhelpful error:
```
Installing dependencies...
  Aborting installation.
    yarn install has failed.
```

### Describe the solution you'd like

Instead of giving that error, there can be 3 approaches:

1. Tell the user exactly that `yarn` isn't found on their system, so either set up the project using `npm`. Just make it like: `Press Y to setup with npm instead`, or they can quit and restart with `npm`.
2. the Second approach would be to say: `Command yarn not found. Press 1 to install it, or press 2 to set up with npm instead.`
3. Third way would be to simply install `yarn` automatically. Just put a message saying `Installing yarn...`

### Describe alternatives you've considered

The alternative is to understand why `yarn install` would fail, which could be because of many reasons I guess. Then figure out that you don't have `yarn` installed, so install it first, then go back to the Turborepo setup.

### PR DESCRIPTION
**Pull request by seanyusa**
Closes #241

Disable the yarn option if not installed.

**Before**
Screen Shot 2021-12-13 at 4 58 13 PM

**After**
Screen Shot 2021-12-13 at 5 02 35 PM

### Note
There is one "valid" path that this prevents. ("valid" as in path that currently works)
That path is:
- user doesn't have yarn installed
- they select "Yarn" for package manager
- but then answer No for "Do you want me to run `yarn install`?"

This path works because we don't have to execute `yarn install` on behalf of the user. Is this a case we want to keep? If so, then we will need to force "No" for the next question instead and log that `yarn install` was not run because `yarn` itself is not installed.

### CODE REVIEW

File: create-turbo/src/yarn.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { execSync } from "child_process";
+
+export function checkIfYarnInstalled(): boolean {
```
Comment (jaredpalmer) [inline create-turbo/src/yarn.ts:3]: Can we reuse create-turbo/src/shouldUseYarn.ts?

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: create-turbo/src/yarn.ts (hunk 1 above)
Comment (seanyusa) [reply create-turbo/src/yarn.ts:3]: @jaredpalmer Ah yes, it's perfect! I didn't look into what the `userAgent` was, and I thought that was checking for something else, learned about a new value today!

Thoughts on renaming `shouldUseYarn`? Reads a bit to me like "are we supposed to use yarn", as opposed to checking if yarn is running/available, which is partially why I didn't look more closely at it in the first place.

Comment (jaredpalmer) [reply create-turbo/src/yarn.ts:3]: Keep it for now.

**Comment by seanyusa** [comment]
Updated! Thanks for reviewing!

**Pull request merged**